File: src/main/java/at/hrechny/predictionsbot/database/entity/TeamEntity.java
package at.hrechny.predictionsbot.database.entity;

import io.micronaut.core.annotation.Introspected;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Index;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;

@Introspected(accessKind = Introspected.AccessKind.FIELD)
@Getter
@Setter
@Entity
@Table(name = "teams", indexes = {
  @Index(name = "idx_api_football_id", columnList = "apiFootballId", unique = true)
})
public class TeamEntity extends GeneratedIdEntity {

  @Column
  private String name;

  @Column
  private Long apiFootballId;

  @Column
  private String logoUrl;

}

File: src/main/kotlin/at/hrechny/predictionsbot/database/entity/TeamEntity.kt
package at.hrechny.predictionsbot.database.entity

import io.micronaut.core.annotation.Introspected
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Table

@Introspected(accessKind = [Introspected.AccessKind.FIELD])
@Entity
@Table(name = "teams")
class TeamEntity : GeneratedIdEntity() {
    @field:Column
    var name: String? = null

    @field:Column
    var logoUrl: String? = null
}
